File: plugins/streaming-skills-plugin/skills/kafka-schema-registry/evals/mock-repos/notification-service/NotificationProducer.cs
using Confluent.Kafka;
using Newtonsoft.Json;
using NotificationService.Models;

namespace NotificationService
{
    public class NotificationProducer
    {
        private readonly IProducer<string, string> _producer;

        public NotificationProducer()
        {
            var config = new ProducerConfig
            {
                BootstrapServers = "localhost:9092"
            };

            _producer = new ProducerBuilder<string, string>(config).Build();
        }

        public async Task SendNotification(UserNotification notification)
        {
            var message = new Message<string, string>
            {
                Key = notification.NotificationId,
                Value = JsonConvert.SerializeObject(notification)
            };

            await _producer.ProduceAsync("user-notifications", message);
        }

        public async Task SendSmsNotification(UserNotification notification)
        {
            var message = new Message<string, string>
            {
                Key = notification.NotificationId,
                Value = JsonConvert.SerializeObject(notification)
            };

            await _producer.ProduceAsync("sms-notifications", message);
        }
    }
}

File: plugins/streaming-skills-plugin/skills/kafka-schema-registry/evals/mock-repos/ecommerce-platform/notification-service/src/notification-producer.js
const { Kafka } = require('kafkajs');

const kafka = new Kafka({
  clientId: 'notification-service',
  brokers: ['localhost:9092']
});

const producer = kafka.producer();

async function sendNotification(notification) {
  await producer.connect();

  await producer.send({
    topic: 'notifications',
    messages: [
      {
        key: notification.notificationId,
        value: JSON.stringify({
          notification_id: notification.notificationId,
          user_email: notification.userEmail,
          user_phone: notification.userPhone,
          message: notification.message,
          sent_at: notification.sentAt
        })
      }
    ]
  });
}

module.exports = { sendNotification };
